modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Semigroupal.kt

```diff
@@ -0,0 +1,122 @@
+package arrow.typeclasses
+
+import arrow.Kind
+import arrow.core.*
+import arrow.documented
+
+/**
+ * ank_macro_hierarchy(arrow.typeclasses.Semigroupal)
+ *
+ * The [Semigroupal] type class for a given type `F` can be seen as an abstraction over the [cartesian product](https://en.wikipedia.org/wiki/Cartesian_product).
+ * It defines the functions [identity] and [product].
+ *
+ * The [identity] function returns the type `Kind<F, A>` for a given type `A`. For example, the identity element of the
+ * type `Kind<ForOption, Int>` aka `ForOption<Int>` is [Option.empty].
+ *
+ * The [product] function for a given type `F`, `A` and `B` combines a `Kind<F, A>` and a `Kind<F, B>` into a `Kind<F, Tuple2<A, B>>`.
+ * This function guarantees compliance with the following laws:
+ *
+ * ```kotlin
+ * identity.product(b) == identity
+ * ```
+ * ```kotlin
+ * a.product(identity) == identity
+ * ```
+ *
+ * [Semigroupal]s are also associative under the bijection `f = (a,(b,c)) -> ((a,b),c)` or `f = ((a,b),c) -> (a,(b,c))`.
+ * Therefore, the following laws also apply:
+ *
+ * ```kotlin
+ * f((a.product(b)).product(c)) == a.product(b.product(c))
+ * ```
+ *
+ * ```kotlin
+ * f(a.product(b.product(c))) == (a.product(b)).product(c)
+ * ```
+ *
+ * Currently, [Semigroupal] instances are defined for [Option], [ListK], [SequenceK] and [SetK].
+ *
+ * ```kotlin:ank:playground:extension
+ * _imports_
+ *
+ * fun main(args: Array<String>) {
+ *   val result =
+ *   //sampleStart
+ *   _extensionFactory_
+ *   //sampleEnd
+ *   println(result)
+ * }
+ * ```
+ *
+ * ### Examples
+ *
+ * Here a some examples:
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.Option
+ * import arrow.core.extensions.option.semigroupal.semigroupal
+ *
+ * fun main(args: Array<String>) {
+ *   val result =
+ *   //sampleStart
+ *   Option.semigroupal().run {
+ *       Option.just(1).product(Option.just(1))
+ *   }
+ *   //sampleEnd
+ *   println(result)
+ * }
+ * ```
+ *
+ * [Semigroupal] also has support of the `*` syntax:
+ *
+ * ```kotlin:ank:playground
+ * import arrow.core.Option
+ * import arrow.core.extensions.option.semigroupal.semigroupal
+ *
+ * fun main(args: Array<String>) {
+ *   val result =
+ *   //sampleStart
+ *   Option.semigroupal().run {
+ *       Option.just(1) * Option.just(2)
+ *   }
+ *   //sampleEnd
+ *   println(result)
+ * }
+ * ```
+ * The same applies to [ListK], [SequenceK] and [SetK] instances:
+ *
+ * ```kotlin:ank:playground
+ * import arrow.data.ListK
+ * import arrow.data.extensions.listk.semigroupal.semigroupal
+ * import arrow.data.k
+ *
+ * fun main(args: Array<String>) {
+ *   val result =
+ *   //sampleStart
+ *   ListK.semigroupal().run {
+ *       listOf(1,2,3).k() * listOf('a','b','c').k()
+ *   }
+ *   //sampleEnd
+ *   println(result)
+ * }
+ * ```
+ */
+@documented
+interface Semigroupal<F> {
+
+    /**
+     * Given a type [A], create an "identity" for a F<A> value.
+     */
+    fun <A> identity(): Kind<F, A>
```

Comment:
Shouldn't this be in `Monoidal` instead?

Reply:
Yeah right. I think I added it into the Semigroupal type class because I misunderstood your comment [here](https://github.com/arrow-kt/arrow/pull/951#issuecomment-407012895). 
I will create a `Monoidal` in a new PR, k?